### DISCUSSION TITLE
how to set up the emscripten toolchain

### DISCUSSION THREAD

**Discussion by Birch-san** [Birch-san/box2d-wasm#16] [Q&A]
Spun out from @hutzlibu's question:  
https://github.com/Birch-san/box2d-wasm/issues/9#issuecomment-778661562

**Comment by Birch-san** [comment]
which OS are you building this on?

**Comment by Birch-san** [comment]
I'd recommend installing latest Emscripten (it worked for me on 2.0.5).

Emscripten 2.0.0 and higher [removed support for the old `fastcomp` backend](https://emscripten.org/docs/compiling/WebAssembly.html#backends), and use an LLVM backend instead. Thus you'll need at least LLVM 12, because emscripten [needs support for `-fignore-exceptions`](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/emscripten.rb#L41). But Emscripten seems to come bundled with its own LLVM distribution, so maybe all you need to do here is "install latest Emscripten".

I think `brew install emscripten` or `brew upgrade emscripten` failed for me in some way (didn't pour the LLVM bottle, or poured me one which didn't work). Instead I installed Emscripten using: `brew reinstall --build-from-source emscripten`. This takes a few hours (compiling LLVM).

**Comment by Birch-san** [comment]
I'd recommend:

- Clone this repository using --recursive, to ensure that the [`box2d`](https://github.com/erincatto/box2d) submodule is available.
  - https://github.com/Birch-san/box2d-wasm/blob/master/README.dev.md#clone-repository
- Compile `box2d` and `Box2D.js` with `TARGET_TYPE=Debug`
  - https://github.com/Birch-san/box2d-wasm/blob/master/box2d-wasm/README.dev.md  
- Run the `integration-test` against your `Box2D.js`
  - https://github.com/Birch-san/box2d-wasm/tree/master/integration-test

**Comment by hutzlibu** [comment]
I use Manjaro, a Arch Linux distribution.
Emscripten  2.0.13-1 is in the repo and installing right now. And I see how far I get this evening. I rather not recompile the whole thing, but ... we'll see ;)
